src/Mod/Draft/draftobjects/label.py

```diff
@@ -0,0 +1,233 @@
+
+        obj.addProperty("App::PropertyStringList",
+                        "Text",
+                        "Base",
+                        QT_TRANSLATE_NOOP("App::Property",
+                                          "The text displayed by this label"))
+
+        obj.StraightDirection = ["Horizontal","Vertical","Custom"]
+        obj.LabelType = ["Custom","Name","Label","Position",
+                         "Length","Area","Volume","Tag","Material"]
+        obj.setEditorMode("Text",1)
+        obj.StraightDistance = 1
+        obj.TargetPoint = App.Vector(2,-1,0)
+        obj.CustomText = "Label"
+
+
+    def execute(self,obj):
+
+        if obj.StraightDirection != "Custom":
+            p1 = obj.Placement.Base
+            if obj.StraightDirection == "Horizontal":
+                p2 = App.Vector(obj.StraightDistance.Value,0,0)
+            else:
+                p2 = App.Vector(0,obj.StraightDistance.Value,0)
+            p2 = obj.Placement.multVec(p2)
+            # p3 = obj.Placement.multVec(obj.TargetPoint)
+            p3 = obj.TargetPoint
+            obj.Points = [p1,p2,p3]
+        if obj.LabelType == "Custom":
+            if obj.CustomText:
+                obj.Text = obj.CustomText
+        elif obj.Target and obj.Target[0]:
+            if obj.LabelType == "Name":
+                obj.Text = [obj.Target[0].Name]
+            elif obj.LabelType == "Label":
+                obj.Text = [obj.Target[0].Label]
+            elif obj.LabelType == "Tag":
+                if hasattr(obj.Target[0],"Tag"):
+                    obj.Text = [obj.Target[0].Tag]
+            elif obj.LabelType == "Material":
+                if hasattr(obj.Target[0],"Material"):
+                    if hasattr(obj.Target[0].Material,"Label"):
+                        obj.Text = [obj.Target[0].Material.Label]
+            elif obj.LabelType == "Position":
+                p = obj.Target[0].Placement.Base
+                if obj.Target[1]:
+                    if "Vertex" in obj.Target[1][0]:
+                        p = obj.Target[0].Shape.Vertexes[int(obj.Target[1][0][6:])-1].Point
+                obj.Text = [App.Units.Quantity(x,App.Units.Length).UserString for x in tuple(p)]
+            elif obj.LabelType == "Length":
+                if hasattr(obj.Target[0],'Shape'):
+                    if hasattr(obj.Target[0].Shape,"Length"):
+                        obj.Text = [App.Units.Quantity(obj.Target[0].Shape.Length,App.Units.Length).UserString]
+                    if obj.Target[1] and ("Edge" in obj.Target[1][0]):
+                        obj.Text = [App.Units.Quantity(obj.Target[0].Shape.Edges[int(obj.Target[1][0][4:])-1].Length,App.Units.Length).UserString]
+            elif obj.LabelType == "Area":
+                if hasattr(obj.Target[0],'Shape'):
+                    if hasattr(obj.Target[0].Shape,"Area"):
+                        obj.Text = [App.Units.Quantity(obj.Target[0].Shape.Area,App.Units.Area).UserString.replace("^2","²")]
```

Comment:
This small 2 is also causing codification problems. `# -*- coding: utf8 -*-`

Reply:
thx for spotting it!